### PR TITLE
chore: infrastructure for injecting scripts into DOMWorlds

### PR DESCRIPTION
**Pull request by jrandolf-google**
This PR removes the `generated` folder in favor of on-demand file generation. This includes improvements to the build system using a GNU Makefile-like scripts (`job.ts`) which drastically improves compilation speed.

This PR also adds the `injected` folder which contains scripts that will be executed in the puppeteer context. Within the `injected` folder, some `Poller`s which will be used in the upcoming refactoring of the wait tasks.

### CODE REVIEW

File: versions.js [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 const versionsPerRelease = new Map([
   // This is a mapping from Chromium version => Puppeteer version.
-  // In Chromium roll patches, use 'NEXT' for the Puppeteer version.
+  // In Chromium roll patches, use 'v16.1.0' for the Puppeteer version.
```
Comment (OrKoN) [inline versions.js]: hm, this is wrong, it should only replace NEXT in the array map

File: src/common/Tracing.ts [hunk 2]
```diff
@@ -13,15 +13,12 @@
...
-  getReadableFromProtocolStream,
-  isErrorLike,
-} from './util.js';
+import {assert} from '../util/assert.js';
+import {getReadableAsBuffer, getReadableFromProtocolStream} from './util.js';
+import {isErrorLike} from '../util/ErrorLike.js';
 import {CDPSession} from './Connection.js';
 
-/**
+/** util/ErrorLike.js
```
Comment (OrKoN) [inline src/common/Tracing.ts]: ?

File: .github/workflows/pre-release.yml [hunk 3]
```diff
@@ -20,7 +20,7 @@ jobs:
         run: npm install
       - name: Build
         run: |
-          node utils/generate_version_file.js
+          ts-node utils/generate_sources.ts
```
Comment (OrKoN) [inline .github/workflows/pre-release.yml:23]: would it be better to run `npm run build` here?

Comment (OrKoN) [reply .github/workflows/pre-release.yml:23]: or at least `npm run generate:sources`

Comment (jrandolf) [reply .github/workflows/pre-release.yml:23]: We just need the `version.js` file that is generated

File: src/common/IsolatedWorld.ts [hunk 4]
```diff
@@ -722,9 +724,7 @@ export class IsolatedWorld {
       waitForVisible: boolean,
       waitForHidden: boolean
     ): Promise<Node | null | boolean> {
-      const node = predicateQueryHandler
-        ? ((await predicateQueryHandler(root, selector)) as Element)
-        : root.querySelector(selector);
+      const node = (await predicateQueryHandler(root, selector)) as Element;
```
Comment (OrKoN) [inline src/common/IsolatedWorld.ts:727]: I think predicateQueryHandler might undefined? If we are sure it's always there, let's update times in makePredicateString to not allow empty predicate query handlers

File: utils/generate_docs.ts [hunk 5]
```diff
@@ -20,8 +20,6 @@ import {chdir} from 'process';
 import semver from 'semver';
 import {versionsPerRelease} from '../versions.js';
 import versionsArchived from '../website/versionsArchived.json';
-
-// eslint-disable-next-line import/extensions
 import {generateDocs} from './internal/custom_markdown_action';
```
Comment (OrKoN) [inline utils/generate_docs.ts:23]: why don't we use a .js extension?

Comment (OrKoN) [reply utils/generate_docs.ts:23]: because of ts-node?

Comment (jrandolf) [reply utils/generate_docs.ts:23]: Yes.

**Review by OrKoN — APPROVED** [review]

**Pull request merged**